File: src/main/java/com/github/yukulab/blockhideandseekmod/command/Reload.java
package com.github.yukulab.blockhideandseekmod.command;

import com.github.yukulab.blockhideandseekmod.BlockHideAndSeekMod;
import com.mojang.brigadier.Command;
import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback;
import net.minecraft.text.Text;

import static net.minecraft.server.command.CommandManager.literal;

public class Reload {
    public static void registerCommands() {
        CommandRegistrationCallback.EVENT.register(
                (dispatcher, dedicated) ->
                        dispatcher.register(literal("bhas")
                                .then(literal("reload")
                                        .requires(source -> source.hasPermissionLevel(BlockHideAndSeekMod.SERVER.getOpPermissionLevel()))
                                        .executes(context -> {
                                            var source = context.getSource();
                                            if (Start.isGameRunning(source)) {
                                                source.sendError(Text.of("[Bhas] ゲーム実行中はリロードできません"));
                                                return Command.SINGLE_SUCCESS;
                                            }
                                            BlockHideAndSeekMod.CONFIG.load();
                                            source.sendFeedback(Text.of("[Bhas] 設定ファイルをリロードしました"), true);
                                            return Command.SINGLE_SUCCESS;
                                        })
                                )
                        )
        );
    }
}

File: src/main/java/com/github/yukulab/blockhideandseekmod/command/Reload.kt
package com.github.yukulab.blockhideandseekmod.command

import com.github.yukulab.blockhideandseekmod.command.BHASCommands.bhasMessage
import com.github.yukulab.blockhideandseekmod.config.Config
import com.github.yukulab.blockhideandseekmod.game.GameController
import dev.uten2c.cmdlib.CommandBuilder
import net.minecraft.text.Text

object Reload : BHASCommand {

    override val builder: CommandBuilder.() -> Unit = {
        literal("reload") {
            requires {
                it.hasPermissionLevel(it.server.opPermissionLevel)
            }

            executes {
                if (!GameController.isGameRunning) {
                    Config.reload()
                    source.sendFeedback(bhasMessage(Text.of("設定ファイルをリロードしました")), true)
                }
            }
        }
    }
}